route/teams.js

```diff
@@ -0,0 +1,233 @@
+  const vname = venue ? venue.name : team.venue;
+
+  let teamRating = 0;
+
+  const lineup = team.roster.map(p => {
+    const rating = IPR.forName(p.name) || 0;
+    // TODO: Handle cases where rating == undefined, instead of default to 0.
+    teamRating += parseInt(rating);
+
+    const pk = makeKey(p.name); //TODO: UGLY makeKey call.
+    // TODO: What divisions to show? Thinking either all or the div of the team.
+    const ps = stats.get(pk).divisions.all;
+    const pops = ps ? ps.points.won / ps.points.of : 0;
+    return {
+      key: pk,
+      name: p.name,
+      rating: rating,
+      pops: ps.pops,
+      points: ps.points.won,
+      num_matches: ps.num_matches,
+      ppm: ps.ppm
+    };
+  }).sort(byName);
+
+  const weeks = team.schedule.map(w => {
+    const isHome = w.side == 'vs';
+    const match = matches.get(w.match_key);
+    let result = null;
+    if(match) {
+      const points = match.getPoints();
+      //result -> '(W|L) team.score - opp.score'
+      if(match.isDone()) {
+        result = isHome ?
+          ((points.home > points.away) ? 'W' : 'L') :
+          ((points.away > points.home) ? 'W' : 'L');
+      }
+      else {
+        result = 'R' + match.round;
+      }
+      result += ' ';
+      const ts = isHome ? points.home : points.away;
+      const os = isHome ? points.away : points.home;
+      result += ts + '-' +os;
+    }
+
+    const [year, month, day] = [
+      w.date.slice(0, 4),
+      w.date.slice(4, 6),
+      w.date.slice(6, 8),
+    ];
+
+    return {
+      week: w.week,
+      date: `${month}-${day}-${year}`,
+      side: w.side,
+      opp:  w.opp,
+      key: w.match_key,
+      result: result
+    };
+  });
+
+  const html = mustache.render(base,{
+    canAdd: req.user.isLeagueAdmin,
+    canRemove: req.user.isLeagueAdmin,
+    team_id: team.key,
+    title: team.name,
+    name: team.name,
+    division: team.division,
+    venue: vname,
+    captain: team.captain,
+    co_captain: team.co_captain,
+    team_rating: teamRating,
+    roster: lineup,
+    schedule: weeks,
+    sugs: JSON.stringify(getSuggestions(), null, 2),
+  }, {
+    content: template
+  });
+
+  res.send(html);
+});
+
+router.post('/teams/:team_id/roster/add', function(req,res) {
+  // TODO: Who is trying to do this, and are they allowed?
+
+  const team = getTeam(req.params.team_id);
+  // TODO: It might be better to return a 500 or something, but since we
+  //  are returning rendered html, that's not as good a model.
+  if(!team) { return res.redirect('/teams'); }
+
+  if(!req.body || !req.body.name || !req.body.role) {
+    // TODO: redirect to /teams/:team_id, but with an error msg.
+    console.log('Jacked up or missing body');
+  }
+
+  // TODO: Do we need to lookup the player in some way?
+  //  If the player wasn't in the suggestions, it may be
+  //  difficult to validate anyways.
+
+  const { name, role } = req.body;
+
+  // TODO: What happens if role is already taken?
```

Comment:
Testing shows that the role is overwritten, but the matching player is not removed. I really like that flow and will leave that in place.